llama.cpp

```diff
@@ -6606,6 +6560,24 @@ struct llama_context_params llama_context_default_params() {
     return result;
 }
 
+struct llama_context_params llama_context_default_params() {
+    struct llama_context_params result = {
+        /*.seed                        =*/ LLAMA_DEFAULT_SEED,
+        /*.n_ctx                       =*/ 512,
+        /*.n_batch                     =*/ 512,
+        /*.n_threads                   =*/ GGML_DEFAULT_N_THREADS, // TODO: better default
+        /*.n_threads_batch             =*/ GGML_DEFAULT_N_THREADS,
```

Comment:
Should we change this to something like `std::thread::hardware_concurrency()`, or `get_num_physical_cores()` from `common.cpp`? It shouldn't affect most of our examples, but a better default could help downstream users.